src/compas_timber/ghpython/components/CT_JointDef_XHalfLap/code.py

```diff
@@ -0,0 +1,40 @@
+from ghpythonlib.componentbase import executingcomponent as component
+from Grasshopper.Kernel.GH_RuntimeMessageLevel import Error
+from Grasshopper.Kernel.GH_RuntimeMessageLevel import Warning
+
+from compas_timber.connections import ConnectionSolver
+from compas_timber.connections import XHalfLapJoint
+from compas_timber.connections import JointTopology
+from compas_timber.ghpython import JointDefinition
+
+
+class XHalfLapDefinition(component):
+    def RunScript(self, MainBeam, CrossBeam):
+        max_distance_from_beams = max([MainBeam[0].width, MainBeam[0].height, CrossBeam[0].width, CrossBeam[0].height])
+        if not MainBeam:
+            self.AddRuntimeMessage(Warning, "Input parameter MainBeams failed to collect data.")
+        if not CrossBeam:
+            self.AddRuntimeMessage(Warning, "Input parameter CrossBeams failed to collect data.")
+        if not (MainBeam and CrossBeam):
+            return
+        if not isinstance(MainBeam, list):
+            MainBeam = [MainBeam]
+        if not isinstance(CrossBeam, list):
+            CrossBeam = [CrossBeam]
+        if len(MainBeam) != len(CrossBeam):
+            self.AddRuntimeMessage(Error, "Number of items in MainBeams and CrossBeams must match!")
+            return
+
+        Joint = []
+        for main, cross in zip(MainBeam, CrossBeam):
+            topology, _, _ = ConnectionSolver().find_topology(main, cross, max_distance = max_distance_from_beams)
```

Comment:
code style: no space around the `=` here. 
```suggestion
            topology, _, _ = ConnectionSolver().find_topology(main, cross, max_distance=max_distance_from_beams)
```